robottelo/cli/factory.py

```diff
@@ -3635,3 +3715,377 @@ def add_role_permissions(role_id, resource_permissions):
+    upload_file(
+        local_file=local_key_path,
+        remote_file=remote_key_path,
+        hostname=vm.ip_addr
+    )
+    result = vm.run('chmod 600 {0}'.format(remote_key_path))
+    if result.return_code != 0:
+        raise CLIFactoryError(
+            u'Failed to chmod ssh key file:\n{}'.format(result.stderr))
+    result = vm.run('touch {0}/config'.format(remote_ssh_path))
+    if result.return_code != 0:
+        raise CLIFactoryError(
+            u'Failed to create ssh config file:\n{}'
+            .format(result.stderr)
+        )
+    if access_hostname:
+        result = vm.run(
+            'echo "\nHost {0}\n\tHostname {0}\n\tUser {1}\n'
+            '\tIdentityFile {2}" >> {3}/config'
+            .format(
+                access_hostname, access_user, remote_key_path, remote_ssh_path)
+        )
+        if result.return_code != 0:
+            raise CLIFactoryError(
+                u'Failed to write to ssh config file:\n{}'
+                .format(result.stderr)
+            )
+        result = vm.run(
+            'ssh-keyscan {0} >> {1}/known_hosts'
+            .format(access_hostname, remote_ssh_path)
+        )
+        if result.return_code != 0:
+            raise CLIFactoryError(
+                u'Failed to put hostname in ssh known_hosts files:\n{}'
+                .format(result.stderr)
+            )
+
+
+def virt_who_hypervisor_config(
+        config_id, virt_who_vm, org_id=None, lce_id=None,
+        hypervisor_hostname=None, configure_ssh=False, hypervisor_user=None,
+        subscription_name=None, exec_one_shot=False, upload_manifest=True):
+    """
+    Configure virtual machine as hypervisor virt-who service
+
+    :param int config_id: virt-who config id
+    :param robottelo.vm.VirtualMachine virt_who_vm: the Virtual machine
+        instance to use for configuration
+    :param int org_id: the organization id
+    :param int lce_id: the lifecycle environment id to use
+    :param str hypervisor_hostname: the hypervisor hostname
+    :param str hypervisor_user: hypervisor user that connect with the ssh key
+    :param bool configure_ssh: whether to configure the ssh key to allow this
+        virtual machine to connect to hypervisor
+    :param str subscription_name: the subscription name to assign to virt-who
+        hypervisor guests
+    :param bool exec_one_shot: whether to run the virt-who one-shot command
+        after startup
+    :param bool upload_manifest: whether to upload the organization manifest
+    """
+    if org_id is None:
+        org = make_org()
+    else:
+        org = Org.info({'id': org_id})
+
+    if lce_id is None:
+        lce = make_lifecycle_environment({'organization-id': org['id']})
+    else:
+        lce = LifecycleEnvironment.info({
+            'id': lce_id,
+            'organization-id': org['id']
+        })
+    rh_product_arch = REPOS['rhel7']['arch']
+    rh_product_releasever = REPOS['rhel7']['releasever']
+    repos = [
```

Comment:
It is a top level factory helper and usually we do not put static constant to them. Do we expect to use these two repos no matter what? Do we plan to use both of them all the time?

Reply:
@oshtaier we can also create a helper function that return the necessary repos that depend from the distro used as we have done for capsule, but any way this also has to be updated in case of many distro changes/usage